### PR TITLE
Rishabh2

### PR DESCRIPTION
**Pull request by RishCSU**

### CODE REVIEW

**Review by TitanSri — COMMENTED** [review]
Rename looks good

File: BRLS/src/library/borrowitem/BorrowItemControl.java [hunk 1]
```diff
@@ -100,17 +105,20 @@ public void borrowingCompleted() {
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemControl.java]: This is missing closing brackets as well however the final changes show it. All the brackets look fine.

File: BRLS/src/library/borrowitem/BorrowItemControl.java [hunk 2]
```diff
@@ -78,8 +78,8 @@ public void itemScanned(int itemId) {
 		}
 
 		pendingList.add(item);
-		for (Item ItEm : pendingList) {
-			ui.DiSpLaY(ItEm);
+		for (Item item : pendingList) {
+			ui.DiSpLaY(item);
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemControl.java]: variable inside the body looks fine

File: BRLS/src/library/borrowitem/BorrowItemControl.java [hunk 3]
```diff
@@ -8,126 +8,126 @@
...
+        state = ControlState.COMPLETED;
+    }
+
+    
+    public void cancel() {
+        ui.setCancelled();
+        state = ControlState.CANCELLED;
+    }
+    
+
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemControl.java:132]: Indentation looks fine

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 4]
```diff
@@ -4,24 +4,24 @@
 
 public class BorrowItemUI {
 	
-	public static enum uI_STaTe { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+	public static enum BorrowItemUIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java]: Spaces at the start of the enum brackets and the end, please fix it. Same with the other class BorrowItemControl

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 5]
```diff
@@ -4,24 +4,24 @@
...
 
-	private uI_STaTe StaTe;
-	private BorrowItemControl CoNtRoL;
-	private Scanner ScAnNeR;
+	private BorrowItemUIState uiState;
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java]: variable names look fine

File: BRLS/src/library/Main.java [hunk 6]
```diff
@@ -159,7 +159,7 @@ private static void LIST_PATRONS() {
 
 
 	private static void BORROW_ITEM() {
-		new BorrowItemUI(new BorrowItemControl()).RuN();		
+		new BorrowItemUI(new BorrowItemControl()).run();
```
Comment (TitanSri) [inline BRLS/src/library/Main.java:162]: methods and parameters looks fine

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 7]
```diff
@@ -43,14 +43,15 @@ public void run() {
 
 				
 			case READY:
-				String PAT_STR = getInput("Swipe patron card (press <enter> to cancel): ");
-				if (PAT_STR.length() == 0) {
+				String patronString = getInput("Swipe patron card (press <enter> to cancel): ");
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java]: the body variable look fine

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 8]
```diff
@@ -52,8 +52,7 @@ public void run() {
 				try {
 					long patronId = Long.valueOf(patronString).longValue();
 					control.cardSwiped(patronId);
-				}
-				catch (NumberFormatException e) {
+				} catch (NumberFormatException e) {
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java]: brackets look fine

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 9]
```diff
@@ -3,136 +3,136 @@
...
+        uiState = BorrowItemUIState.FINALISING;
+    }
+
+    public void setCompleted() {
+        uiState = BorrowItemUIState.COMPLETED;
+    }
+
+    public void setCancelled() {
+        uiState = BorrowItemUIState.CANCELLED;
+    }
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java:136]: Indentation looks fine

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 10]
```diff
@@ -3,147 +3,136 @@
...
-
+    
+    public static enum BorrowItemUIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+    private BorrowItemUIState uiState;
+    private BorrowItemControl control;
+    private Scanner scanner;
+
+    
+    public BorrowItemUI(BorrowItemControl control) {
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java:14]: parameter should be borrowItemControl not control, please fix it

**Review by TitanSri — CHANGES_REQUESTED** [review]
The enum spaces inside the bracket need to be fixed on both classes and the constructor parameter for BorrowItemUI, please fix

File: BRLS/src/library/borrowitem/BorrowItemUI.java [hunk 11]
```diff
@@ -3,147 +3,136 @@
...
+    
+                
+    public void run() {
+        displayOutput("Borrow Item Use Case UI\n");
+        
+        while (true) {
+            
+            switch (uiState) {            
+            
+            case CANCELLED:
```
Comment (TitanSri) [inline BRLS/src/library/borrowitem/BorrowItemUI.java:40]: All the case indentations have to be incremented 4 spaces and the body to align with the case

**Review by TitanSri — CHANGES_REQUESTED** [review]
Must fix the case and body of the case indentation for the BorrowItemUI,java file

**Comment by DChand2021-2023** [comment]
Rishabh, please address the issues identified by Toby.
 Thanks, Deepak

File: BRLS/src/library/borrowitem/BorrowItemControl.java [hunk 12]
```diff
@@ -13,8 +13,8 @@ public class BorrowItemControl {
 	
 	private Library lIbRaRy;
 	private Patron PaTrOn;
-	private enum CONTROL_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
-	private CONTROL_STATE sTaTe;
+	private enum ControlState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
```
[the remark this answers is not in the document]
Comment (RishCSU) [reply BRLS/src/library/borrowitem/BorrowItemControl.java]: I have rectified this mistake

File: BRLS/src/library/borrowitem/BorrowItemUI.java (hunk 4 above)
Comment (RishCSU) [reply BRLS/src/library/borrowitem/BorrowItemUI.java]: I have rectified this mistake

File: BRLS/src/library/borrowitem/BorrowItemUI.java (hunk 10 above)
Comment (RishCSU) [reply BRLS/src/library/borrowitem/BorrowItemUI.java:14]: I have rectified this mistake

File: BRLS/src/library/borrowitem/BorrowItemUI.java (hunk 11 above)
Comment (RishCSU) [reply BRLS/src/library/borrowitem/BorrowItemUI.java]: I have rectified this mistake

File: BRLS/src/library/borrowitem/BorrowItemControl.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (TitanSri) [reply BRLS/src/library/borrowitem/BorrowItemControl.java]: Thanks defect has been fixed by removing the spaces

File: BRLS/src/library/borrowitem/BorrowItemUI.java (hunk 4 above)
Comment (TitanSri) [reply BRLS/src/library/borrowitem/BorrowItemUI.java]: Thank you for removing the spaces

File: BRLS/src/library/borrowitem/BorrowItemUI.java (hunk 10 above)
Comment (TitanSri) [reply BRLS/src/library/borrowitem/BorrowItemUI.java]: Parameter name has been changed to the class diagram, thank you

File: BRLS/src/library/borrowitem/BorrowItemUI.java (hunk 11 above)
Comment (TitanSri) [reply BRLS/src/library/borrowitem/BorrowItemUI.java]: Indentation was added, Good job.

**Review by TitanSri — APPROVED** [review]
All 4 defects has been fixed

**Review by ghost — APPROVED** [review]
Tested as a mediator and approved.

**Comment by RishCSU** [comment]
So glad to work with you guys :)

**Pull request merged**